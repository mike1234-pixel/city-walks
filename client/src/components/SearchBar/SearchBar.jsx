import { MDBInput, MDBBtn } from "mdbreact"
import { FaSearchLocation } from "react-icons/fa"
import './SeachBar.css'

let SearchBar = (props) => {

    return (
      <form onSubmit={props.handleSubmit}>
        <span style={{ marginRight: "40px"}} className="search-container">
          <MDBInput  
            data-testid="search-input" 
            label="Search" 
            name="search-input" 
            id="search-input" 
            type="text" placeholder="Search" 
            value={props.searchValue} 
            onChange={props.handleChange}
            />
         <button type="submit" data-testid="search-btn" className="search-btn"><FaSearchLocation id="search-btn-icon" className="search-location-icon"/></button> 
        </span>
      </form>
    )
}

export default SearchBar;

// controlled input:
// --> input value is set with *props* provided through React 
// --> the form data and the input value is updated through on *onChange* handler


