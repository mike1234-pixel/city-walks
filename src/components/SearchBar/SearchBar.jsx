let SearchBar = (props) => {

    return (
      <form onSubmit={props.handleSubmit}>
          <input  name="search-input" id="search-input" type="text" placeholder="Search" value={props.searchValue} onChange={props.handleChange}/>
          <button type="submit">Search</button>
      </form>
    )
}

export default SearchBar;

// controlled input:
// --> input value is set with *props* provided through React 
// --> the form data and the input value is updated through on *onChange* handler