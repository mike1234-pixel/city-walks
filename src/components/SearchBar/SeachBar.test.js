import React from 'react'
import renderer from 'react-test-renderer'
import SearchBar from './SearchBar'

// snapshot test
// jest will create a snapshot automatically on first run
// to update the snapshot run npm test -- --updateSnapshot
it('SearchBar renders correctly', () => {  
  const tree = renderer
    .create(<SearchBar/>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});




