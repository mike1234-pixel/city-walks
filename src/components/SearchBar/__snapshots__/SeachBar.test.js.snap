// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar renders correctly 1`] = `
<form>
  <input
    id="search-input"
    name="search-input"
    placeholder="Search"
    type="text"
  />
  <button
    type="submit"
  >
    Search
  </button>
</form>
`;
